File: iniciante-beginner/c/1049.c
/**
 * Source: Beecrowd
 * @problem: 1049 - Animal
 * @subject: Selection
 *
 * @author: Letícia Cruz
 * @date: 2020-08-23
 */

#include <stdio.h>
#include <string.h>

int main(void)
{
    char subfilo[15], classe[15], alimentacao[15];
    scanf("%s\n %s\n %s", subfilo, classe, alimentacao);

    if (strcmp(subfilo, "vertebrado") == 0)
    {
        if (strcmp(classe, "ave") == 0)
        {
            if (strcmp(alimentacao, "carnivoro") == 0)
            {
                printf("aguia\n");
            }
            else
            {
                printf("pomba\n");
            }
        }
        else
        {
            if (strcmp(alimentacao, "onivoro") == 0)
            {
                printf("homem\n");
            }
            else
            {
                printf("vaca\n");
            }
        }
    }
    else
    {
        if (strcmp(classe, "inseto") == 0)
        {
            if (strcmp(alimentacao, "hematofago") == 0)
            {
                printf("pulga\n");
            }
            else
            {
                printf("lagarta\n");
            }
        }
        else
        {
            if (strcmp(alimentacao, "hematofago") == 0)
            {
                printf("sanguessuga\n");
            }
            else
            {
                printf("minhoca\n");
            }
        }
    }

    return 0;
}

File: iniciante-beginner/javascript/1049.js
/**
 * Source: Beecrowd
 * @problem: 1049 - Animal
 * @subject: Selection
 *
 * @author: Letícia Cruz
 * @date: 2020-11-13
 */

var input = require("fs").readFileSync("/dev/stdin", "utf8");
var lines = input.split(/\r?\n/);

let categoria1 = lines.shift();
let categoria2 = lines.shift();
let categoria3 = lines.shift();

let animal;

if (categoria1 === "vertebrado") {
    if (categoria2 === "ave") {
        if (categoria3 === "carnivoro") {
            animal = "aguia";
        } else {
            animal = "pomba";
        }
    } else {
        if (categoria3 === "onivoro") {
            animal = "homem";
        } else {
            animal = "vaca";
        }
    }
} else {
    if (categoria2 === "inseto") {
        if (categoria3 === "hematofago") {
            animal = "pulga";
        } else {
            animal = "lagarta";
        }
    } else {
        if (categoria3 === "hematofago") {
            animal = "sanguessuga";
        } else {
            animal = "minhoca";
        }
    }
}

console.log(animal);
